helix-view/src/editor.rs

```diff
@@ -1333,4 +1354,23 @@ impl Editor {
             }
         }
     }
+
+    pub async fn flush_writes(&mut self) {
+        while self.write_count > 0 {
+            if let Some(save_event) = self.save_queue.next().await {
+                match &save_event {
+                    Ok(event) => {
+                        let doc = doc_mut!(self, &event.doc_id);
+                        doc.set_last_saved_revision(event.revision);
+                    }
+                    Err(err) => {
+                        log::error!("error saving document: {}", err);
+                    }
+                };
+                // TODO: if is_err: break?
```

Comment:
Yeah we should return an error. From an API perspective, I think it makes more sense to let the caller decide how to handle an error. I can take care of this.